// @builder.io/sdk-angular/lib/browser/esm2022/functions/get-processed-block.mjs
import { TARGET } from '../constants/target';
import { omit } from '../helpers/omit';
import { evaluate } from './evaluate/index';
import { resolveLocalizedValues } from './extract-localized-values';
import { fastClone } from './fast-clone';
import { set } from './set';
import { transformBlock } from './transform-block';
// Deep clone a block but without cloning any child blocks
export function deepCloneWithConditions(obj) {
    if (obj === null || typeof obj !== 'object') {
        return obj;
    }
    if (Array.isArray(obj)) {
        return obj.map((item) => deepCloneWithConditions(item));
    }
    if (obj['@type'] === '@builder.io/sdk:Element') {
        return obj;
    }
    const clonedObj = {};
    for (const key in obj) {
        if (key !== 'meta' && Object.prototype.hasOwnProperty.call(obj, key)) {
            clonedObj[key] = deepCloneWithConditions(obj[key]);
        }
    }
    return clonedObj;
}
const IS_SDK_WITHOUT_CACHED_PROCESSED_BLOCK = ['svelte', 'vue', 'angular', 'qwik', 'solid'].includes(TARGET);
const getCopy = (block) => {
    if (IS_SDK_WITHOUT_CACHED_PROCESSED_BLOCK) {
        const copy = fastClone(block);
        const copied = {
            ...copy,
            properties: {
                ...copy.properties
            },
            actions: {
                ...copy.actions
            }
        };
        return copied;
    }
    else {
        const copy = deepCloneWithConditions(omit(block, 'children', 'meta'));
        return {
            ...copy,
            properties: {
                ...copy.properties
            },
            actions: {
                ...copy.actions
            },
            children: block.children,
            meta: block.meta
        };
    }
};
const evaluateBindings = ({ block, context, localState, rootState, rootSetState }) => {
    if (!block.bindings) {
        return block;
    }
    const copied = getCopy(block);
    for (const binding in block.bindings) {
        const expression = block.bindings[binding];
        const value = evaluate({
            code: expression,
            localState,
            rootState,
            rootSetState,
            context
        });
        set(copied, binding, value);
    }
    return copied;
};
export function getProcessedBlock({ block, context, localState, rootState, rootSetState }) {
    let transformedBlock = resolveLocalizedValues(block, rootState.locale);
    transformedBlock = transformBlock(transformedBlock);
    return evaluateBindings({
        block: transformedBlock,
        localState,
        rootState,
        rootSetState,
        context
    });
}
