// @builder.io/sdk-angular/lib/browser/esm2022/context/builder.context.mjs
import { Injectable } from "@angular/core";
import * as i0 from "@angular/core";
export default class BuilderContext {
    rootSetState() { }
    constructor() {
        this.content = null;
        this.context = {};
        this.localState = undefined;
        this.rootState = {};
        this.apiKey = null;
        this.apiVersion = undefined;
        this.componentInfos = {};
        this.inheritedStyles = {};
        this.BlocksWrapper = "div";
        this.BlocksWrapperProps = {};
        this.nonce = "";
        this.model = "";
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderContext, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderContext, providedIn: "root" }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderContext, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: "root",
                }]
        }], ctorParameters: function () { return []; } });
