// @builder.io/sdk-angular/lib/browser/esm2022/blocks/button/button.mjs
import { Component, Input } from "@angular/core";
import { CommonModule } from "@angular/common";
import DynamicRenderer from "../../components/dynamic-renderer/dynamic-renderer";
import { getClassPropName } from "../../functions/get-class-prop-name";
import * as i0 from "@angular/core";
export default class BuilderButton {
    constructor() {
        this.node_0_DynamicRenderer = null;
        this.node_1_DynamicRenderer = null;
    }
    attrs() {
        return {
            ...this.attributes,
            [getClassPropName()]: `${this.link ? "" : "builder-button"} ${this.attributes[getClassPropName()] || ""}`,
            ...(this.link
                ? {
                    href: this.link,
                    target: this.openLinkInNewTab ? "_blank" : undefined,
                    role: "link",
                }
                : {
                    role: "button",
                }),
        };
    }
    ngOnInit() {
        this.node_0_DynamicRenderer = this.attrs();
        this.node_1_DynamicRenderer = {};
    }
    ngOnChanges(changes) {
        if (typeof window !== "undefined") {
            this.node_0_DynamicRenderer = this.attrs();
            this.node_1_DynamicRenderer = {};
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderButton, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: BuilderButton, isStandalone: true, selector: "builder-button", inputs: { attributes: "attributes", link: "link", openLinkInNewTab: "openLinkInNewTab", builderLinkComponent: "builderLinkComponent", text: "text" }, usesOnChanges: true, ngImport: i0, template: `
    <dynamic-renderer
      [attributes]="node_0_DynamicRenderer"
      [TagName]="link ? builderLinkComponent || 'a' : 'button'"
      [actionAttributes]="node_1_DynamicRenderer"
    >
      {{text}}
    </dynamic-renderer>
  `, isInline: true, styles: [":host{display:contents}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "component", type: DynamicRenderer, selector: "dynamic-renderer, DynamicRenderer", inputs: ["TagName", "attributes", "actionAttributes"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderButton, decorators: [{
            type: Component,
            args: [{ selector: "builder-button", template: `
    <dynamic-renderer
      [attributes]="node_0_DynamicRenderer"
      [TagName]="link ? builderLinkComponent || 'a' : 'button'"
      [actionAttributes]="node_1_DynamicRenderer"
    >
      {{text}}
    </dynamic-renderer>
  `, standalone: true, imports: [CommonModule, DynamicRenderer], styles: [":host{display:contents}\n"] }]
        }], propDecorators: { attributes: [{
                type: Input
            }], link: [{
                type: Input
            }], openLinkInNewTab: [{
                type: Input
            }], builderLinkComponent: [{
                type: Input
            }], text: [{
                type: Input
            }] } });
