// @builder.io/sdk-angular/lib/node/esm2022/components/content/components/styles.mjs
import { Component, Input } from "@angular/core";
import { CommonModule } from "@angular/common";
import InlinedStyles from "../../inlined-styles";
import { getCss, getDefaultStyles, getFontCss } from "./styles.helpers";
import * as i0 from "@angular/core";
export default class ContentStyles {
    constructor() {
        this.injectedStyles = null;
    }
    ngOnInit() {
        this.injectedStyles = `
${getCss({
            cssCode: this.cssCode,
            contentId: this.contentId,
        })}
${getFontCss({
            customFonts: this.customFonts,
        })}
${getDefaultStyles(this.isNestedRender)}
`.trim();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: ContentStyles, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: ContentStyles, isStandalone: true, selector: "content-styles", inputs: { cssCode: "cssCode", contentId: "contentId", customFonts: "customFonts", isNestedRender: "isNestedRender", nonce: "nonce" }, ngImport: i0, template: `
    <inlined-styles
      id="builderio-content"
      [styles]="injectedStyles"
      [nonce]="nonce"
    ></inlined-styles>
  `, isInline: true, styles: [":host{display:contents}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "component", type: InlinedStyles, selector: "inlined-styles, InlinedStyles", inputs: ["styles", "id", "nonce"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: ContentStyles, decorators: [{
            type: Component,
            args: [{ selector: "content-styles", template: `
    <inlined-styles
      id="builderio-content"
      [styles]="injectedStyles"
      [nonce]="nonce"
    ></inlined-styles>
  `, standalone: true, imports: [CommonModule, InlinedStyles], styles: [":host{display:contents}\n"] }]
        }], propDecorators: { cssCode: [{
                type: Input
            }], contentId: [{
                type: Input
            }], customFonts: [{
                type: Input
            }], isNestedRender: [{
                type: Input
            }], nonce: [{
                type: Input
            }] } });
