// @builder.io/sdk-angular/lib/node/esm2022/blocks/text/text.mjs
import { ChangeDetectionStrategy } from "@angular/core";
import { Component, Input } from "@angular/core";
import { CommonModule } from "@angular/common";
import * as i0 from "@angular/core";
import * as i1 from "@angular/platform-browser";
import * as i2 from "@angular/common";
export default class BuilderText {
    ngOnInit() {
        this.node_0_div = this.text?.toString() || "";
        this.node_1_div = {
            outline: "none",
        };
    }
    constructor(sanitizer) {
        this.sanitizer = sanitizer;
        this.node_0_div = null;
        this.node_1_div = null;
    }
    ngOnChanges(changes) {
        if (typeof window !== "undefined") {
            this.node_0_div = this.text?.toString() || "";
            this.node_1_div = {
                outline: "none",
            };
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderText, deps: [{ token: i1.DomSanitizer }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: BuilderText, isStandalone: true, selector: "builder-text", inputs: { text: "text" }, usesOnChanges: true, ngImport: i0, template: `
    <div
      class="builder-text"
      [innerHTML]="sanitizer.bypassSecurityTrustHtml(node_0_div)"
      [ngStyle]="node_1_div"
    ></div>
  `, isInline: true, styles: [":host{display:contents}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i2.NgStyle, selector: "[ngStyle]", inputs: ["ngStyle"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderText, decorators: [{
            type: Component,
            args: [{ changeDetection: ChangeDetectionStrategy.OnPush, selector: "builder-text", template: `
    <div
      class="builder-text"
      [innerHTML]="sanitizer.bypassSecurityTrustHtml(node_0_div)"
      [ngStyle]="node_1_div"
    ></div>
  `, standalone: true, imports: [CommonModule], styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i1.DomSanitizer }]; }, propDecorators: { text: [{
                type: Input
            }] } });
