// @builder.io/sdk-angular/lib/node/esm2022/components/content/content.mjs
import { Component, Input } from "@angular/core";
import { CommonModule } from "@angular/common";
import { getDefaultRegisteredComponents } from "../../constants/builder-registered-components";
import { TARGET } from "../../constants/target";
import { evaluate } from "../../functions/evaluate/evaluate";
import { serializeIncludingFunctions } from "../../functions/register-component";
import { logger } from "../../helpers/logger";
import Blocks from "../blocks/blocks";
import { getUpdateVariantVisibilityScript } from "../content-variants/helpers";
import DynamicDiv from "../dynamic-div";
import InlinedScript from "../inlined-script";
import EnableEditor from "./components/enable-editor";
import ContentStyles from "./components/styles";
import { getContentInitialValue, getRootStateInitialValue, } from "./content.helpers";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export default class ContentComponent {
    constructor() {
        this.TARGET = TARGET;
        this.scriptStr = null;
        this.registeredComponents = null;
        this.builderContextSignal = null;
    }
    contentSetState(newRootState) {
        this.builderContextSignal.rootState = newRootState;
    }
    ngOnInit() {
        this.scriptStr = getUpdateVariantVisibilityScript({
            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion, @typescript-eslint/no-non-null-asserted-optional-chain
            variationId: this.content?.testVariationId,
            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion, @typescript-eslint/no-non-null-asserted-optional-chain
            contentId: this.content?.id,
        });
        this.registeredComponents = [
            ...getDefaultRegisteredComponents(),
            ...(this.customComponents || []),
        ].reduce((acc, { component, ...info }) => ({
            ...acc,
            [info.name]: {
                component: component,
                ...serializeIncludingFunctions(info),
            },
        }), {});
        this.builderContextSignal = {
            content: getContentInitialValue({
                content: this.content,
                data: this.data,
            }),
            localState: undefined,
            rootState: getRootStateInitialValue({
                content: this.content,
                data: this.data,
                locale: this.locale,
            }),
            rootSetState: this.contentSetState.bind(this),
            context: this.context || {},
            canTrack: this.canTrack,
            apiKey: this.apiKey,
            apiVersion: this.apiVersion,
            componentInfos: [
                ...getDefaultRegisteredComponents(),
                ...(this.customComponents || []),
            ].reduce((acc, { component: _, ...info }) => ({
                ...acc,
                [info.name]: serializeIncludingFunctions(info),
            }), {}),
            inheritedStyles: {},
            BlocksWrapper: this.blocksWrapper || DynamicDiv,
            BlocksWrapperProps: this.blocksWrapperProps || {},
            nonce: this.nonce || "",
            model: this.model,
        };
        if (!this.apiKey) {
            logger.error("No API key provided to `Content` component. This can cause issues. Please provide an API key using the `apiKey` prop.");
        }
        // run any dynamic JS code attached to content
        const jsCode = this.builderContextSignal.content?.data?.jsCode;
        if (jsCode) {
            evaluate({
                code: jsCode,
                context: this.context || {},
                localState: undefined,
                rootState: this.builderContextSignal.rootState,
                rootSetState: (newState) => {
                    this.builderContextSignal.rootSetState?.(newState);
                },
                isExpression: false,
            });
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: ContentComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: ContentComponent, isStandalone: true, selector: "content-component", inputs: { apiKey: "apiKey", context: "context", content: "content", customComponents: "customComponents", data: "data", locale: "locale", canTrack: "canTrack", apiVersion: "apiVersion", blocksWrapper: "blocksWrapper", blocksWrapperProps: "blocksWrapperProps", nonce: "nonce", model: "model", apiHost: "apiHost", enrich: "enrich", showContent: "showContent", contentWrapper: "contentWrapper", contentWrapperProps: "contentWrapperProps", trustedHosts: "trustedHosts", isNestedRender: "isNestedRender", isSsrAbTest: "isSsrAbTest", linkComponent: "linkComponent" }, host: { attributes: { "ngSkipHydration": "true" } }, ngImport: i0, template: `
    <enable-editor
      [apiHost]="apiHost"
      [nonce]="nonce"
      [content]="content"
      [data]="data"
      [model]="model"
      [context]="context"
      [apiKey]="apiKey"
      [canTrack]="canTrack"
      [locale]="locale"
      [enrich]="enrich"
      [showContent]="showContent"
      [builderContextSignal]="builderContextSignal"
      [contentWrapper]="contentWrapper"
      [contentWrapperProps]="contentWrapperProps"
      [trustedHosts]="trustedHosts"
      [isNestedRender]="isNestedRender"
    >
      <ng-container *ngIf="isSsrAbTest">
        <inlined-script
          id="builderio-variant-visibility"
          [scriptStr]="scriptStr"
          [nonce]="nonce || ''"
        ></inlined-script>
      </ng-container>
      <ng-container *ngIf="TARGET !== 'reactNative'">
        <content-styles
          [nonce]="nonce || ''"
          [isNestedRender]="isNestedRender"
          [contentId]="builderContextSignal.content?.id"
          [cssCode]="builderContextSignal.content?.data?.cssCode"
          [customFonts]="builderContextSignal.content?.data?.customFonts"
        ></content-styles>
      </ng-container>
      <blocks
        [blocks]="builderContextSignal.content?.data?.blocks"
        [context]="builderContextSignal"
        [registeredComponents]="registeredComponents"
        [linkComponent]="linkComponent"
      ></blocks>
    </enable-editor>
  `, isInline: true, styles: [":host{display:contents}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "component", type: EnableEditor, selector: "enable-editor", inputs: ["isNestedRender", "locale", "enrich", "trustedHosts", "model", "apiKey", "builderContextSignal", "canTrack", "apiHost", "content", "context", "data", "showContent", "contentWrapper", "contentWrapperProps"] }, { kind: "component", type: InlinedScript, selector: "inlined-script, InlinedScript", inputs: ["scriptStr", "id", "nonce"] }, { kind: "component", type: ContentStyles, selector: "content-styles", inputs: ["cssCode", "contentId", "customFonts", "isNestedRender", "nonce"] }, { kind: "component", type: Blocks, selector: "blocks", inputs: ["blocks", "parent", "path", "styleProp", "className", "context", "linkComponent", "registeredComponents"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: ContentComponent, decorators: [{
            type: Component,
            args: [{ host: { ngSkipHydration: "true" }, selector: "content-component", template: `
    <enable-editor
      [apiHost]="apiHost"
      [nonce]="nonce"
      [content]="content"
      [data]="data"
      [model]="model"
      [context]="context"
      [apiKey]="apiKey"
      [canTrack]="canTrack"
      [locale]="locale"
      [enrich]="enrich"
      [showContent]="showContent"
      [builderContextSignal]="builderContextSignal"
      [contentWrapper]="contentWrapper"
      [contentWrapperProps]="contentWrapperProps"
      [trustedHosts]="trustedHosts"
      [isNestedRender]="isNestedRender"
    >
      <ng-container *ngIf="isSsrAbTest">
        <inlined-script
          id="builderio-variant-visibility"
          [scriptStr]="scriptStr"
          [nonce]="nonce || ''"
        ></inlined-script>
      </ng-container>
      <ng-container *ngIf="TARGET !== 'reactNative'">
        <content-styles
          [nonce]="nonce || ''"
          [isNestedRender]="isNestedRender"
          [contentId]="builderContextSignal.content?.id"
          [cssCode]="builderContextSignal.content?.data?.cssCode"
          [customFonts]="builderContextSignal.content?.data?.customFonts"
        ></content-styles>
      </ng-container>
      <blocks
        [blocks]="builderContextSignal.content?.data?.blocks"
        [context]="builderContextSignal"
        [registeredComponents]="registeredComponents"
        [linkComponent]="linkComponent"
      ></blocks>
    </enable-editor>
  `, standalone: true, imports: [CommonModule, EnableEditor, InlinedScript, ContentStyles, Blocks], styles: [":host{display:contents}\n"] }]
        }], propDecorators: { apiKey: [{
                type: Input
            }], context: [{
                type: Input
            }], content: [{
                type: Input
            }], customComponents: [{
                type: Input
            }], data: [{
                type: Input
            }], locale: [{
                type: Input
            }], canTrack: [{
                type: Input
            }], apiVersion: [{
                type: Input
            }], blocksWrapper: [{
                type: Input
            }], blocksWrapperProps: [{
                type: Input
            }], nonce: [{
                type: Input
            }], model: [{
                type: Input
            }], apiHost: [{
                type: Input
            }], enrich: [{
                type: Input
            }], showContent: [{
                type: Input
            }], contentWrapper: [{
                type: Input
            }], contentWrapperProps: [{
                type: Input
            }], trustedHosts: [{
                type: Input
            }], isNestedRender: [{
                type: Input
            }], isSsrAbTest: [{
                type: Input
            }], linkComponent: [{
                type: Input
            }] } });
