// @builder.io/sdk-angular/lib/browser/esm2022/helpers/user-attributes.mjs
import { isBrowser } from '../functions/is-browser';
import { getCookieSync, setCookie } from './cookie';
export const USER_ATTRIBUTES_COOKIE_NAME = 'builder.userAttributes';
export function createUserAttributesService() {
    let canTrack = true;
    const subscribers = new Set();
    return {
        setUserAttributes(newAttrs) {
            if (!isBrowser()) {
                return;
            }
            const userAttributes = {
                ...this.getUserAttributes(),
                ...newAttrs
            };
            setCookie({
                name: USER_ATTRIBUTES_COOKIE_NAME,
                value: JSON.stringify(userAttributes),
                canTrack
            });
            subscribers.forEach(callback => callback(userAttributes));
        },
        getUserAttributes() {
            if (!isBrowser()) {
                return {};
            }
            return JSON.parse(getCookieSync({
                name: USER_ATTRIBUTES_COOKIE_NAME,
                canTrack
            }) || '{}');
        },
        subscribeOnUserAttributesChange(callback) {
            subscribers.add(callback);
            return () => {
                subscribers.delete(callback);
            };
        },
        setCanTrack(value) {
            canTrack = value;
        }
    };
}
export const userAttributesService = createUserAttributesService();
export const setClientUserAttributes = (attributes) => {
    userAttributesService.setUserAttributes(attributes);
};
