// @builder.io/sdk-angular/lib/browser/esm2022/functions/track/index.mjs
import { TARGET } from '../../constants/target';
import { logger } from '../../helpers/logger';
import { getSdkHeaders } from '../../helpers/sdk-headers';
import { getSessionId } from '../../helpers/sessionId';
import { getVisitorId } from '../../helpers/visitorId';
import { isBrowser } from '../is-browser';
import { isEditing } from '../is-editing';
import { logFetch } from '../log-fetch';
import { getUserAttributes } from './helpers';
const getTrackingEventData = async ({ canTrack }) => {
    if (!canTrack) {
        return {
            visitorId: undefined,
            sessionId: undefined
        };
    }
    const sessionId = await getSessionId({
        canTrack
    });
    const visitorId = getVisitorId({
        canTrack
    });
    return {
        sessionId,
        visitorId
    };
};
const createEvent = async ({ type: eventType, canTrack, apiKey, metadata, ...properties }) => ({
    type: eventType,
    data: {
        ...properties,
        metadata: {
            url: location.href,
            ...metadata
        },
        ...(await getTrackingEventData({
            canTrack
        })),
        userAttributes: getUserAttributes(),
        ownerId: apiKey
    }
});
export async function _track({ apiHost, ...eventProps }) {
    if (!eventProps.apiKey) {
        logger.error('Missing API key for track call. Please provide your API key.');
        return;
    }
    if (!eventProps.canTrack) {
        return;
    }
    if (isEditing()) {
        return;
    }
    if (!(isBrowser() || TARGET === 'reactNative')) {
        return;
    }
    const baseUrl = apiHost || 'https://cdn.builder.io';
    const url = `${baseUrl}/api/v1/track`;
    logFetch(url);
    return fetch(url, {
        method: 'POST',
        body: JSON.stringify({
            events: [await createEvent(eventProps)]
        }),
        headers: {
            'content-type': 'application/json',
            ...getSdkHeaders()
        },
        mode: 'cors'
    }).catch(err => {
        console.error('Failed to track: ', err);
    });
}
export const track = (args) => _track({
    ...args,
    canTrack: true
});
