// @builder.io/sdk-angular/lib/node/esm2022/components/dynamic-renderer/dynamic-renderer.mjs
import { CommonModule } from '@angular/common';
import { Component, ElementRef, Input, ViewChild } from '@angular/core';
import { isEmptyElement } from './dynamic-renderer.helpers';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export default class DynamicRenderer {
    useTypeOf(obj) {
        return typeof obj;
    }
    constructor(vcRef) {
        this.vcRef = vcRef;
        this.isEmptyElement = isEmptyElement;
    }
    ngOnInit() {
        if (typeof this.TagName === 'string') {
            switch (this.TagName) {
                case 'a':
                    this.TagName = DynamicA;
                    break;
                case 'button':
                    this.TagName = DynamicButton;
                    break;
                case 'div':
                    this.TagName = DynamicDiv;
                    break;
                case 'span':
                    this.TagName = DynamicSpan;
                    break;
                case 'p':
                    this.TagName = DynamicP;
                    break;
                case 'img':
                    this.TagName = DynamicImg;
                    break;
                case 'input':
                    this.TagName = DynamicInput;
                    break;
                case 'textarea':
                    this.TagName = DynamicTextarea;
                    break;
                case 'select':
                    this.TagName = DynamicSelect;
                    break;
                case 'option':
                    this.TagName = DynamicOption;
                    break;
                case 'form':
                    this.TagName = DynamicForm;
                    break;
                case 'label':
                    this.TagName = DynamicLabel;
                    break;
                case 'ul':
                    this.TagName = DynamicUl;
                    break;
                case 'li':
                    this.TagName = DynamicLi;
                    break;
                case 'table':
                    this.TagName = DynamicTable;
                    break;
                case 'tr':
                    this.TagName = DynamicTr;
                    break;
                case 'td':
                    this.TagName = DynamicTd;
                    break;
                case 'th':
                    this.TagName = DynamicTh;
                    break;
                case 'thead':
                    this.TagName = DynamicThead;
                    break;
                case 'tbody':
                    this.TagName = DynamicTbody;
                    break;
                case 'footer':
                    this.TagName = DynamicFooter;
                    break;
                case 'header':
                    this.TagName = DynamicHeader;
                    break;
                case 'nav':
                    this.TagName = DynamicNav;
                    break;
                case 'section':
                    this.TagName = DynamicSection;
                    break;
                case 'article':
                    this.TagName = DynamicArticle;
                    break;
                case 'aside':
                    this.TagName = DynamicAside;
                    break;
                case 'h1':
                    this.TagName = DynamicH1;
                    break;
                case 'h2':
                    this.TagName = DynamicH2;
                    break;
                case 'h3':
                    this.TagName = DynamicH3;
                    break;
                case 'h4':
                    this.TagName = DynamicH4;
                    break;
                case 'h5':
                    this.TagName = DynamicH5;
                    break;
                case 'h6':
                    this.TagName = DynamicH6;
                    break;
                case 'blockquote':
                    this.TagName = DynamicBlockquote;
                    break;
                case 'code':
                    this.TagName = DynamicCode;
                    break;
                case 'pre':
                    this.TagName = DynamicPre;
                    break;
                case 'figure':
                    this.TagName = DynamicFigure;
                    break;
                case 'figcaption':
                    this.TagName = DynamicFigcaption;
                    break;
                case 'video':
                    this.TagName = DynamicVideo;
                    break;
                case 'audio':
                    this.TagName = DynamicAudio;
                    break;
                case 'canvas':
                    this.TagName = DynamicCanvas;
                    break;
                case 'iframe':
                    this.TagName = DynamicIframe;
                    break;
                default:
                    break;
            }
        }
        this.myContent = [this.vcRef.createEmbeddedView(this.tagnameTemplateRef).rootNodes];
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicRenderer, deps: [{ token: i0.ViewContainerRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicRenderer, isStandalone: true, selector: "dynamic-renderer, DynamicRenderer", inputs: { TagName: "TagName", attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "tagnameTemplateRef", first: true, predicate: ["tagnameTemplate"], descendants: true, static: true }], ngImport: i0, template: `
    <ng-template #tagnameTemplate><ng-content></ng-content></ng-template>
    <ng-container *ngIf="!isEmptyElement(TagName)">
      <ng-container *ngIf="useTypeOf(TagName) === 'string'">
        <ng-container
          *ngComponentOutlet="
            TagName;
            inputs: {
              attributes: attributes,
              actionAttributes: actionAttributes
            };
            content: myContent
          "
        ></ng-container>
      </ng-container>
      <ng-container *ngIf="!(useTypeOf(TagName) === 'string')">
        <ng-container
          *ngComponentOutlet="
            TagName;
            inputs: {
              attributes: attributes,
              actionAttributes: actionAttributes
            };
            content: myContent
          "
        ></ng-container>
      </ng-container>
    </ng-container>
    <ng-container *ngIf="!!isEmptyElement(TagName)">
      <ng-container
        *ngComponentOutlet="
          TagName;
          inputs: {
            attributes: attributes,
            actionAttributes: actionAttributes
          };
          content: myContent
        "
      ></ng-container>
    </ng-container>
  `, isInline: true, styles: [":host{display:contents}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgComponentOutlet, selector: "[ngComponentOutlet]", inputs: ["ngComponentOutlet", "ngComponentOutletInputs", "ngComponentOutletInjector", "ngComponentOutletContent", "ngComponentOutletNgModule", "ngComponentOutletNgModuleFactory"] }, { kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicRenderer, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-renderer, DynamicRenderer', template: `
    <ng-template #tagnameTemplate><ng-content></ng-content></ng-template>
    <ng-container *ngIf="!isEmptyElement(TagName)">
      <ng-container *ngIf="useTypeOf(TagName) === 'string'">
        <ng-container
          *ngComponentOutlet="
            TagName;
            inputs: {
              attributes: attributes,
              actionAttributes: actionAttributes
            };
            content: myContent
          "
        ></ng-container>
      </ng-container>
      <ng-container *ngIf="!(useTypeOf(TagName) === 'string')">
        <ng-container
          *ngComponentOutlet="
            TagName;
            inputs: {
              attributes: attributes,
              actionAttributes: actionAttributes
            };
            content: myContent
          "
        ></ng-container>
      </ng-container>
    </ng-container>
    <ng-container *ngIf="!!isEmptyElement(TagName)">
      <ng-container
        *ngComponentOutlet="
          TagName;
          inputs: {
            attributes: attributes,
            actionAttributes: actionAttributes
          };
          content: myContent
        "
      ></ng-container>
    </ng-container>
  `, standalone: true, imports: [CommonModule], styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }]; }, propDecorators: { TagName: [{
                type: Input
            }], attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], tagnameTemplateRef: [{
                type: ViewChild,
                args: ['tagnameTemplate', { static: true }]
            }] } });
export class DynamicA {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicA, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicA, isStandalone: true, selector: "dynamic-a, DynamicA", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <a #v ><ng-content></ng-content></a> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicA, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-a, DynamicA', template: ` <a #v ><ng-content></ng-content></a> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicButton {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicButton, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicButton, isStandalone: true, selector: "dynamic-button, DynamicButton", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <button #v ><ng-content></ng-content></button> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicButton, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-button, DynamicButton', template: ` <button #v ><ng-content></ng-content></button> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicDiv {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicDiv, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicDiv, isStandalone: true, selector: "dynamic-div, DynamicDiv", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <div #v ><ng-content></ng-content></div> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicDiv, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-div, DynamicDiv', template: ` <div #v ><ng-content></ng-content></div> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicSpan {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicSpan, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicSpan, isStandalone: true, selector: "dynamic-span, DynamicSpan", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <span #v ><ng-content></ng-content></span> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicSpan, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-span, DynamicSpan', template: ` <span #v ><ng-content></ng-content></span> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicP {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicP, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicP, isStandalone: true, selector: "dynamic-p, DynamicP", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <p #v ><ng-content></ng-content></p> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicP, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-p, DynamicP', template: ` <p #v ><ng-content></ng-content></p> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicImg {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicImg, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicImg, isStandalone: true, selector: "dynamic-img, DynamicImg", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <img #v /> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicImg, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-img, DynamicImg', template: ` <img #v /> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicInput {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicInput, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicInput, isStandalone: true, selector: "dynamic-input, DynamicInput", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <input #v /> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicInput, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-input, DynamicInput', template: ` <input #v /> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicTextarea {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicTextarea, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicTextarea, isStandalone: true, selector: "dynamic-textarea, DynamicTextarea", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <textarea #v ><ng-content></ng-content></textarea> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicTextarea, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-textarea, DynamicTextarea', template: ` <textarea #v ><ng-content></ng-content></textarea> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicSelect {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicSelect, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicSelect, isStandalone: true, selector: "dynamic-select, DynamicSelect", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <select #v ><ng-content></ng-content></select> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicSelect, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-select, DynamicSelect', template: ` <select #v ><ng-content></ng-content></select> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicOption {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicOption, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicOption, isStandalone: true, selector: "dynamic-option, DynamicOption", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <option #v ><ng-content></ng-content></option> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicOption, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-option, DynamicOption', template: ` <option #v ><ng-content></ng-content></option> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicForm {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicForm, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicForm, isStandalone: true, selector: "dynamic-form, DynamicForm", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <form #v ><ng-content></ng-content></form> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicForm, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-form, DynamicForm', template: ` <form #v ><ng-content></ng-content></form> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicLabel {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicLabel, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicLabel, isStandalone: true, selector: "dynamic-label, DynamicLabel", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <label #v ><ng-content></ng-content></label> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicLabel, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-label, DynamicLabel', template: ` <label #v ><ng-content></ng-content></label> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicUl {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicUl, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicUl, isStandalone: true, selector: "dynamic-ul, DynamicUl", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <ul #v ><ng-content></ng-content></ul> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicUl, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-ul, DynamicUl', template: ` <ul #v ><ng-content></ng-content></ul> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicLi {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicLi, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicLi, isStandalone: true, selector: "dynamic-li, DynamicLi", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <li #v ><ng-content></ng-content></li> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicLi, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-li, DynamicLi', template: ` <li #v ><ng-content></ng-content></li> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicTable {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicTable, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicTable, isStandalone: true, selector: "dynamic-table, DynamicTable", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <table #v ><ng-content></ng-content></table> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicTable, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-table, DynamicTable', template: ` <table #v ><ng-content></ng-content></table> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicTr {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicTr, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicTr, isStandalone: true, selector: "dynamic-tr, DynamicTr", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <tr #v ><ng-content></ng-content></tr> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicTr, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-tr, DynamicTr', template: ` <tr #v ><ng-content></ng-content></tr> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicTd {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicTd, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicTd, isStandalone: true, selector: "dynamic-td, DynamicTd", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <td #v ><ng-content></ng-content></td> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicTd, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-td, DynamicTd', template: ` <td #v ><ng-content></ng-content></td> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicTh {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicTh, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicTh, isStandalone: true, selector: "dynamic-th, DynamicTh", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <th #v ><ng-content></ng-content></th> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicTh, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-th, DynamicTh', template: ` <th #v ><ng-content></ng-content></th> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicThead {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicThead, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicThead, isStandalone: true, selector: "dynamic-thead, DynamicThead", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <thead #v ><ng-content></ng-content></thead> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicThead, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-thead, DynamicThead', template: ` <thead #v ><ng-content></ng-content></thead> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicTbody {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicTbody, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicTbody, isStandalone: true, selector: "dynamic-tbody, DynamicTbody", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <tbody #v ><ng-content></ng-content></tbody> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicTbody, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-tbody, DynamicTbody', template: ` <tbody #v ><ng-content></ng-content></tbody> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicFooter {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicFooter, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicFooter, isStandalone: true, selector: "dynamic-footer, DynamicFooter", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <footer #v ><ng-content></ng-content></footer> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicFooter, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-footer, DynamicFooter', template: ` <footer #v ><ng-content></ng-content></footer> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicHeader {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicHeader, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicHeader, isStandalone: true, selector: "dynamic-header, DynamicHeader", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <header #v ><ng-content></ng-content></header> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicHeader, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-header, DynamicHeader', template: ` <header #v ><ng-content></ng-content></header> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicNav {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicNav, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicNav, isStandalone: true, selector: "dynamic-nav, DynamicNav", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <nav #v ><ng-content></ng-content></nav> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicNav, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-nav, DynamicNav', template: ` <nav #v ><ng-content></ng-content></nav> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicSection {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicSection, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicSection, isStandalone: true, selector: "dynamic-section, DynamicSection", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <section #v ><ng-content></ng-content></section> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicSection, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-section, DynamicSection', template: ` <section #v ><ng-content></ng-content></section> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicArticle {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicArticle, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicArticle, isStandalone: true, selector: "dynamic-article, DynamicArticle", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <article #v ><ng-content></ng-content></article> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicArticle, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-article, DynamicArticle', template: ` <article #v ><ng-content></ng-content></article> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicAside {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicAside, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicAside, isStandalone: true, selector: "dynamic-aside, DynamicAside", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <aside #v ><ng-content></ng-content></aside> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicAside, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-aside, DynamicAside', template: ` <aside #v ><ng-content></ng-content></aside> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicH1 {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicH1, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicH1, isStandalone: true, selector: "dynamic-h1, DynamicH1", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <h1 #v ><ng-content></ng-content></h1> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicH1, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-h1, DynamicH1', template: ` <h1 #v ><ng-content></ng-content></h1> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicH2 {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicH2, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicH2, isStandalone: true, selector: "dynamic-h2, DynamicH2", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <h2 #v ><ng-content></ng-content></h2> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicH2, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-h2, DynamicH2', template: ` <h2 #v ><ng-content></ng-content></h2> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicH3 {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicH3, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicH3, isStandalone: true, selector: "dynamic-h3, DynamicH3", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <h3 #v ><ng-content></ng-content></h3> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicH3, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-h3, DynamicH3', template: ` <h3 #v ><ng-content></ng-content></h3> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicH4 {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicH4, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicH4, isStandalone: true, selector: "dynamic-h4, DynamicH4", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <h4 #v ><ng-content></ng-content></h4> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicH4, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-h4, DynamicH4', template: ` <h4 #v ><ng-content></ng-content></h4> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicH5 {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicH5, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicH5, isStandalone: true, selector: "dynamic-h5, DynamicH5", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <h5 #v ><ng-content></ng-content></h5> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicH5, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-h5, DynamicH5', template: ` <h5 #v ><ng-content></ng-content></h5> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicH6 {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicH6, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicH6, isStandalone: true, selector: "dynamic-h6, DynamicH6", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <h6 #v ><ng-content></ng-content></h6> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicH6, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-h6, DynamicH6', template: ` <h6 #v ><ng-content></ng-content></h6> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicBlockquote {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicBlockquote, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicBlockquote, isStandalone: true, selector: "dynamic-blockquote, DynamicBlockquote", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <blockquote #v ><ng-content></ng-content></blockquote> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicBlockquote, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-blockquote, DynamicBlockquote', template: ` <blockquote #v ><ng-content></ng-content></blockquote> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicCode {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicCode, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicCode, isStandalone: true, selector: "dynamic-code, DynamicCode", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <code #v ><ng-content></ng-content></code> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicCode, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-code, DynamicCode', template: ` <code #v ><ng-content></ng-content></code> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicPre {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicPre, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicPre, isStandalone: true, selector: "dynamic-pre, DynamicPre", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <pre #v ><ng-content></ng-content></pre> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicPre, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-pre, DynamicPre', template: ` <pre #v ><ng-content></ng-content></pre> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicFigure {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicFigure, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicFigure, isStandalone: true, selector: "dynamic-figure, DynamicFigure", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <figure #v ><ng-content></ng-content></figure> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicFigure, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-figure, DynamicFigure', template: ` <figure #v ><ng-content></ng-content></figure> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicFigcaption {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicFigcaption, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicFigcaption, isStandalone: true, selector: "dynamic-figcaption, DynamicFigcaption", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <figcaption #v ><ng-content></ng-content></figcaption> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicFigcaption, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-figcaption, DynamicFigcaption', template: ` <figcaption #v ><ng-content></ng-content></figcaption> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicVideo {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicVideo, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicVideo, isStandalone: true, selector: "dynamic-video, DynamicVideo", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <video #v ><ng-content></ng-content></video> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicVideo, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-video, DynamicVideo', template: ` <video #v ><ng-content></ng-content></video> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicAudio {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicAudio, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicAudio, isStandalone: true, selector: "dynamic-audio, DynamicAudio", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <audio #v ><ng-content></ng-content></audio> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicAudio, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-audio, DynamicAudio', template: ` <audio #v ><ng-content></ng-content></audio> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicCanvas {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicCanvas, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicCanvas, isStandalone: true, selector: "dynamic-canvas, DynamicCanvas", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <canvas #v ><ng-content></ng-content></canvas> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicCanvas, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-canvas, DynamicCanvas', template: ` <canvas #v ><ng-content></ng-content></canvas> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
export class DynamicIframe {
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
    }
    setAttributes(el, value, changes) {
        if (!el)
            return;
        const target = changes ? changes : value;
        Object.keys(target).forEach((key) => {
            if (key.startsWith('on')) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace('on', '').toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key, target[key] ?? '');
            }
        });
    }
    ngAfterViewInit() {
        this.setAttributes(this.v?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (changes.attributes) {
            this.setAttributes(this.v?.nativeElement, this.attributes, changes.attributes.currentValue);
        }
    }
    ngOnDestroy() {
        this._listenerFns.forEach(fn => fn());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicIframe, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DynamicIframe, isStandalone: true, selector: "dynamic-iframe, DynamicIframe", inputs: { attributes: "attributes", actionAttributes: "actionAttributes" }, viewQueries: [{ propertyName: "v", first: true, predicate: ["v"], descendants: true, read: ElementRef }], usesOnChanges: true, ngImport: i0, template: ` <iframe #v ><ng-content></ng-content></iframe> `, isInline: true, styles: [":host{display:contents}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DynamicIframe, decorators: [{
            type: Component,
            args: [{ selector: 'dynamic-iframe, DynamicIframe', template: ` <iframe #v ><ng-content></ng-content></iframe> `, standalone: true, styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { attributes: [{
                type: Input
            }], actionAttributes: [{
                type: Input
            }], v: [{
                type: ViewChild,
                args: ['v', { read: ElementRef }]
            }] } });
