// @builder.io/sdk-angular/lib/browser/esm2022/blocks/video/video.mjs
import { Component, ViewChild, Input, } from "@angular/core";
import { CommonModule } from "@angular/common";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export default class BuilderVideo {
    get videoProps() {
        return {
            ...(this.autoPlay === true
                ? {
                    autoPlay: true,
                }
                : {}),
            ...(this.muted === true
                ? {
                    muted: true,
                }
                : {}),
            ...(this.controls === true
                ? {
                    controls: true,
                }
                : {}),
            ...(this.loop === true
                ? {
                    loop: true,
                }
                : {}),
            ...(this.playsInline === true
                ? {
                    playsInline: true,
                }
                : {}),
        };
    }
    get spreadProps() {
        return {
            ...this.videoProps,
        };
    }
    setAttributes(el, value, changes) {
        if (!el) {
            return;
        }
        const target = typeof changes === "undefined" ? value : changes;
        Object.keys(target).forEach((key) => {
            if (key.startsWith("on")) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace("on", "").toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key.toLowerCase(), target[key] ?? "");
            }
        });
    }
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
        this.lazyVideoObserver = undefined;
        this.node_0_div = null;
        this.node_1_video = null;
        this.node_2_source = null;
        this.node_3_Show = null;
        this.node_4_div = null;
        this.node_5_div = null;
        this.node_6_div = null;
    }
    ngOnInit() {
        this.node_0_div = {
            position: "relative",
        };
        this.node_1_video = {
            width: "100%",
            height: "100%",
            ...this.attributes?.style,
            objectFit: this.fit,
            objectPosition: this.position,
            // Hack to get object fit to work as expected and
            // not have the video overflow
            borderRadius: "1px",
            ...(this.aspectRatio
                ? {
                    position: "absolute",
                }
                : null),
        };
        this.node_2_source = {
            ...(this.lazyLoad
                ? {
                    "data-src": this.video,
                }
                : {
                    src: this.video,
                }),
        };
        this.node_3_Show =
            this.aspectRatio &&
                !(this.fitContent && this.builderBlock?.children?.length);
        this.node_4_div = {
            width: "100%",
            paddingTop: this.aspectRatio * 100 + "%",
            pointerEvents: "none",
            fontSize: "0px",
        };
        this.node_5_div = {
            display: "flex",
            flexDirection: "column",
            alignItems: "stretch",
        };
        this.node_6_div = {
            pointerEvents: "none",
            display: "flex",
            flexDirection: "column",
            alignItems: "stretch",
            position: "absolute",
            top: "0",
            left: "0",
            width: "100%",
            height: "100%",
        };
    }
    ngAfterViewInit() {
        if (this.lazyLoad) {
            const oberver = new IntersectionObserver(function (entries) {
                entries.forEach(function (entry) {
                    if (!entry.isIntersecting)
                        return;
                    const videoElement = entry.target;
                    try {
                        // Convert HTMLCollection to Array and filter for source elements
                        Array.from(videoElement.children)
                            .filter((child) => child instanceof HTMLElement && child.tagName === "SOURCE")
                            .forEach((source) => {
                            const src = source.dataset.src;
                            if (src) {
                                source.src = src;
                            }
                        });
                        videoElement.load();
                        oberver.unobserve(videoElement);
                    }
                    catch (error) {
                        console.error("Error loading lazy video:", error);
                    }
                });
            });
            if (this.videoRef.nativeElement) {
                oberver.observe(this.videoRef.nativeElement);
            }
            this.lazyVideoObserver = oberver;
        }
        this.setAttributes(this.elRef0?.nativeElement, this.spreadProps);
        this.setAttributes(this.elRef1?.nativeElement, this.node_2_source);
    }
    ngOnChanges(changes) {
        if (typeof window !== "undefined") {
            this.node_0_div = {
                position: "relative",
            };
            this.node_1_video = {
                width: "100%",
                height: "100%",
                ...this.attributes?.style,
                objectFit: this.fit,
                objectPosition: this.position,
                // Hack to get object fit to work as expected and
                // not have the video overflow
                borderRadius: "1px",
                ...(this.aspectRatio
                    ? {
                        position: "absolute",
                    }
                    : null),
            };
            this.node_2_source = {
                ...(this.lazyLoad
                    ? {
                        "data-src": this.video,
                    }
                    : {
                        src: this.video,
                    }),
            };
            this.node_3_Show =
                this.aspectRatio &&
                    !(this.fitContent && this.builderBlock?.children?.length);
            this.node_4_div = {
                width: "100%",
                paddingTop: this.aspectRatio * 100 + "%",
                pointerEvents: "none",
                fontSize: "0px",
            };
            this.node_5_div = {
                display: "flex",
                flexDirection: "column",
                alignItems: "stretch",
            };
            this.node_6_div = {
                pointerEvents: "none",
                display: "flex",
                flexDirection: "column",
                alignItems: "stretch",
                position: "absolute",
                top: "0",
                left: "0",
                width: "100%",
                height: "100%",
            };
            this.setAttributes(this.elRef0?.nativeElement, this.spreadProps, changes["spreadProps"]?.currentValue);
            this.setAttributes(this.elRef1?.nativeElement, this.node_2_source, changes["node_2_source"]?.currentValue);
        }
    }
    ngOnDestroy() {
        if (this.lazyVideoObserver) {
            this.lazyVideoObserver.disconnect();
        }
        for (const fn of this._listenerFns.values()) {
            fn();
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderVideo, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: BuilderVideo, isStandalone: true, selector: "builder-video", inputs: { lazyLoad: "lazyLoad", attributes: "attributes", fit: "fit", position: "position", aspectRatio: "aspectRatio", video: "video", fitContent: "fitContent", builderBlock: "builderBlock", autoPlay: "autoPlay", muted: "muted", controls: "controls", loop: "loop", playsInline: "playsInline", preload: "preload", posterImage: "posterImage" }, viewQueries: [{ propertyName: "videoRef", first: true, predicate: ["videoRef"], descendants: true }, { propertyName: "elRef0", first: true, predicate: ["elRef0"], descendants: true }, { propertyName: "elRef1", first: true, predicate: ["elRef1"], descendants: true }], usesOnChanges: true, ngImport: i0, template: `
    <div [ngStyle]="node_0_div">
      <video
        class="builder-video"
        #videoRef
        [attr.preload]="lazyLoad ? 'none' : preload || 'metadata'"
        [ngStyle]="node_1_video"
        [attr.poster]="posterImage"
        #elRef0
      >
        <source type="video/mp4" #elRef1 />
      </video>
      <ng-container *ngIf="node_3_Show">
        <div [ngStyle]="node_4_div"></div>
      </ng-container>
      <ng-container *ngIf="builderBlock?.children?.length && fitContent">
        <div [ngStyle]="node_5_div"><ng-content></ng-content></div>
      </ng-container>
      <ng-container *ngIf="builderBlock?.children?.length && !fitContent">
        <div [ngStyle]="node_6_div"><ng-content></ng-content></div>
      </ng-container>
    </div>
  `, isInline: true, styles: [":host{display:contents}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i1.NgStyle, selector: "[ngStyle]", inputs: ["ngStyle"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderVideo, decorators: [{
            type: Component,
            args: [{ selector: "builder-video", template: `
    <div [ngStyle]="node_0_div">
      <video
        class="builder-video"
        #videoRef
        [attr.preload]="lazyLoad ? 'none' : preload || 'metadata'"
        [ngStyle]="node_1_video"
        [attr.poster]="posterImage"
        #elRef0
      >
        <source type="video/mp4" #elRef1 />
      </video>
      <ng-container *ngIf="node_3_Show">
        <div [ngStyle]="node_4_div"></div>
      </ng-container>
      <ng-container *ngIf="builderBlock?.children?.length && fitContent">
        <div [ngStyle]="node_5_div"><ng-content></ng-content></div>
      </ng-container>
      <ng-container *ngIf="builderBlock?.children?.length && !fitContent">
        <div [ngStyle]="node_6_div"><ng-content></ng-content></div>
      </ng-container>
    </div>
  `, standalone: true, imports: [CommonModule], styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { lazyLoad: [{
                type: Input
            }], attributes: [{
                type: Input
            }], fit: [{
                type: Input
            }], position: [{
                type: Input
            }], aspectRatio: [{
                type: Input
            }], video: [{
                type: Input
            }], fitContent: [{
                type: Input
            }], builderBlock: [{
                type: Input
            }], autoPlay: [{
                type: Input
            }], muted: [{
                type: Input
            }], controls: [{
                type: Input
            }], loop: [{
                type: Input
            }], playsInline: [{
                type: Input
            }], preload: [{
                type: Input
            }], posterImage: [{
                type: Input
            }], videoRef: [{
                type: ViewChild,
                args: ["videoRef"]
            }], elRef0: [{
                type: ViewChild,
                args: ["elRef0"]
            }], elRef1: [{
                type: ViewChild,
                args: ["elRef1"]
            }] } });
