// @builder.io/sdk-angular/lib/node/esm2022/components/inlined-script.mjs
import { CommonModule, DOCUMENT, isPlatformBrowser } from '@angular/common';
// fails because type imports cannot be injected
// eslint-disable-next-line @typescript-eslint/consistent-type-imports
import { Component, Inject, Input, PLATFORM_ID, } from '@angular/core';
import * as i0 from "@angular/core";
export default class InlinedScript {
    constructor(renderer, document, platformId) {
        this.renderer = renderer;
        this.document = document;
        this.platformId = platformId;
    }
    ngOnInit() {
        if (isPlatformBrowser(this.platformId)) {
            const script = this.renderer.createElement('script');
            script.innerHTML = this.scriptStr;
            this.renderer.setAttribute(script, 'data-id', this.id);
            this.renderer.setAttribute(script, 'nonce', this.nonce);
            this.renderer.appendChild(this.document.body, script);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: InlinedScript, deps: [{ token: i0.Renderer2 }, { token: DOCUMENT }, { token: PLATFORM_ID }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: InlinedScript, isStandalone: true, selector: "inlined-script, InlinedScript", inputs: { scriptStr: "scriptStr", id: "id", nonce: "nonce" }, ngImport: i0, template: ``, isInline: true, styles: [":host{display:contents}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: InlinedScript, decorators: [{
            type: Component,
            args: [{ selector: 'inlined-script, InlinedScript', template: ``, standalone: true, imports: [CommonModule], styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }, { type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }]; }, propDecorators: { scriptStr: [{
                type: Input
            }], id: [{
                type: Input
            }], nonce: [{
                type: Input
            }] } });
