// @builder.io/sdk-angular/lib/node/esm2022/components/inlined-styles.mjs
import { CommonModule, DOCUMENT } from '@angular/common';
// fails because type imports cannot be injected
// eslint-disable-next-line @typescript-eslint/consistent-type-imports
import { Component, Inject, Input } from '@angular/core';
import * as i0 from "@angular/core";
export default class InlinedStyles {
    constructor(renderer, document) {
        this.renderer = renderer;
        this.document = document;
    }
    ngOnChanges(changes) {
        if (changes.styles) {
            if (this.styleElement) {
                this.styleElement.textContent = this.styles;
            }
            else {
                this.styleElement = this.renderer.createElement('style');
                this.renderer.setAttribute(this.styleElement, 'data-id', this.id);
                this.renderer.appendChild(this.styleElement, this.renderer.createText(this.styles));
                this.renderer.setAttribute(this.styleElement, 'nonce', this.nonce);
                this.renderer.appendChild(this.document.head, this.styleElement);
            }
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: InlinedStyles, deps: [{ token: i0.Renderer2 }, { token: DOCUMENT }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: InlinedStyles, isStandalone: true, selector: "inlined-styles, InlinedStyles", inputs: { styles: "styles", id: "id", nonce: "nonce" }, usesOnChanges: true, ngImport: i0, template: ``, isInline: true, styles: [":host{display:contents}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: InlinedStyles, decorators: [{
            type: Component,
            args: [{ selector: 'inlined-styles, InlinedStyles', template: ``, standalone: true, imports: [CommonModule], styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }, { type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }]; }, propDecorators: { styles: [{
                type: Input
            }], id: [{
                type: Input
            }], nonce: [{
                type: Input
            }] } });
