// @builder.io/sdk-angular/lib/node/esm2022/blocks/form/form/form.mjs
import { Component, ViewChild, Input, } from "@angular/core";
import { CommonModule } from "@angular/common";
import Blocks from "../../../components/blocks/blocks";
import { getEnv } from "../../../functions/get-env";
import { get } from "../../../functions/get";
import { isEditing } from "../../../functions/is-editing";
import { set } from "../../../functions/set";
import { logFetch } from "../../../functions/log-fetch";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export default class FormComponent {
    mergeNewRootState(newData) {
        const combinedState = {
            ...this.builderContext.rootState,
            ...newData,
        };
        if (this.builderContext.rootSetState) {
            this.builderContext.rootSetState?.(combinedState);
        }
        else {
            this.builderContext.rootState = combinedState;
        }
    }
    submissionState() {
        return (isEditing() && this.previewState) || this.formState;
    }
    onSubmit(event) {
        const sendWithJsProp = this.sendWithJs || this.sendSubmissionsTo === "email";
        if (this.sendSubmissionsTo === "zapier") {
            event.preventDefault();
        }
        else if (sendWithJsProp) {
            if (!(this.action || this.sendSubmissionsTo === "email")) {
                event.preventDefault();
                return;
            }
            event.preventDefault();
            const el = event.currentTarget || event.target;
            const headers = this.customHeaders || {};
            let body;
            const formData = new FormData(el);
            // TODO: maybe support null
            const formPairs = Array.from(el.querySelectorAll("input,select,textarea"))
                .filter((el) => !!el.name)
                .map((el) => {
                let value;
                const key = el.name;
                if (el instanceof HTMLInputElement) {
                    if (el.type === "radio") {
                        if (el.checked) {
                            value = el.name;
                            return {
                                key,
                                value,
                            };
                        }
                    }
                    else if (el.type === "checkbox") {
                        value = el.checked;
                    }
                    else if (el.type === "number" || el.type === "range") {
                        const num = el.valueAsNumber;
                        if (!isNaN(num)) {
                            value = num;
                        }
                    }
                    else if (el.type === "file") {
                        // TODO: one vs multiple files
                        value = el.files;
                    }
                    else {
                        value = el.value;
                    }
                }
                else {
                    value = el.value;
                }
                return {
                    key,
                    value,
                };
            });
            let formContentType = this.contentType;
            if (this.sendSubmissionsTo === "email") {
                formContentType = "multipart/form-data";
            }
            Array.from(formPairs).forEach(({ value }) => {
                if (value instanceof File ||
                    (Array.isArray(value) && value[0] instanceof File) ||
                    value instanceof FileList) {
                    formContentType = "multipart/form-data";
                }
            });
            // TODO: send as urlEncoded or multipart by default
            // because of ease of use and reliability in browser API
            // for encoding the form?
            if (formContentType !== "application/json") {
                body = formData;
            }
            else {
                // Json
                const json = {};
                Array.from(formPairs).forEach(({ value, key }) => {
                    set(json, key, value);
                });
                body = JSON.stringify(json);
            }
            if (formContentType && formContentType !== "multipart/form-data") {
                if (
                /* Zapier doesn't allow content-type header to be sent from browsers */
                !(sendWithJsProp && this.action?.includes("zapier.com"))) {
                    headers["content-type"] = formContentType;
                }
            }
            const presubmitEvent = new CustomEvent("presubmit", {
                detail: {
                    body,
                },
            });
            if (this.formRef.nativeElement) {
                this.formRef.nativeElement.dispatchEvent(presubmitEvent);
                if (presubmitEvent.defaultPrevented) {
                    return;
                }
            }
            this.formState = "sending";
            const formUrl = `${getEnv() === "dev" ? "http://localhost:5000" : "https://builder.io"}/api/v1/form-submit?apiKey=${this.builderContext.apiKey}&to=${btoa(this.sendSubmissionsToEmail || "")}&name=${encodeURIComponent(this.name || "")}`;
            const url = this.sendSubmissionsTo === "email"
                ? formUrl
                : this.action; /* TODO: throw error if no action URL */
            logFetch(url);
            fetch(url, {
                body,
                headers,
                method: this.method || "post",
            }).then(async (res) => {
                let body;
                const contentType = res.headers.get("content-type");
                if (contentType && contentType.indexOf("application/json") !== -1) {
                    body = await res.json();
                }
                else {
                    body = await res.text();
                }
                if (!res.ok && this.errorMessagePath) {
                    /* TODO: allow supplying an error formatter function */
                    let message = get(body, this.errorMessagePath);
                    if (message) {
                        if (typeof message !== "string") {
                            /* TODO: ideally convert json to yaml so it woul dbe like
                        error: - email has been taken */
                            message = JSON.stringify(message);
                        }
                        this.formErrorMessage = message;
                        this.mergeNewRootState({
                            formErrorMessage: message,
                        });
                    }
                }
                this.responseData = body;
                this.formState = res.ok ? "success" : "error";
                if (res.ok) {
                    const submitSuccessEvent = new CustomEvent("submit:success", {
                        detail: {
                            res,
                            body,
                        },
                    });
                    if (this.formRef.nativeElement) {
                        this.formRef.nativeElement.dispatchEvent(submitSuccessEvent);
                        if (submitSuccessEvent.defaultPrevented) {
                            return;
                        }
                        /* TODO: option to turn this on/off? */
                        if (this.resetFormOnSubmit !== false) {
                            this.formRef.nativeElement.reset();
                        }
                    }
                    /* TODO: client side route event first that can be preventDefaulted */
                    if (this.successUrl) {
                        if (this.formRef.nativeElement) {
                            const event = new CustomEvent("route", {
                                detail: {
                                    url: this.successUrl,
                                },
                            });
                            this.formRef.nativeElement.dispatchEvent(event);
                            if (!event.defaultPrevented) {
                                location.href = this.successUrl;
                            }
                        }
                        else {
                            location.href = this.successUrl;
                        }
                    }
                }
            }, (err) => {
                const submitErrorEvent = new CustomEvent("submit:error", {
                    detail: {
                        error: err,
                    },
                });
                if (this.formRef.nativeElement) {
                    this.formRef.nativeElement.dispatchEvent(submitErrorEvent);
                    if (submitErrorEvent.defaultPrevented) {
                        return;
                    }
                }
                this.responseData = err;
                this.formState = "error";
            });
        }
    }
    setAttributes(el, value, changes) {
        if (!el) {
            return;
        }
        const target = typeof changes === "undefined" ? value : changes;
        Object.keys(target).forEach((key) => {
            if (key.startsWith("on")) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace("on", "").toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key.toLowerCase(), target[key] ?? "");
            }
        });
    }
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
        this.formState = "unsubmitted";
        this.responseData = null;
        this.formErrorMessage = "";
        this.node_1_Show = null;
        this.node_2_Show = null;
        this.node_3_Show = null;
        this.node_4_div = null;
        this.node_5_Show = null;
        this.elRef0_state_0 = null;
        this.elRef0_state_1 = null;
    }
    ngOnInit() {
        this.node_1_Show = this.submissionState() === "error";
        this.node_2_Show = this.submissionState() === "sending";
        this.node_3_Show = this.submissionState() === "error" && this.responseData;
        this.node_4_div = JSON.stringify(this.responseData, null, 2);
        this.node_5_Show = this.submissionState() === "success";
        this.elRef0_state_0 = {};
        this.elRef0_state_1 = {};
    }
    ngAfterViewInit() {
        this.setAttributes(this.elRef0?.nativeElement, this.elRef0_state_0);
        this.setAttributes(this.elRef0?.nativeElement, this.elRef0_state_1);
        this.setAttributes(this.elRef0?.nativeElement, this.attributes);
    }
    ngOnChanges(changes) {
        if (typeof window !== "undefined") {
            this.node_1_Show = this.submissionState() === "error";
            this.node_2_Show = this.submissionState() === "sending";
            this.node_3_Show =
                this.submissionState() === "error" && this.responseData;
            this.node_4_div = JSON.stringify(this.responseData, null, 2);
            this.node_5_Show = this.submissionState() === "success";
            this.elRef0_state_0 = {};
            this.setAttributes(this.elRef0?.nativeElement, this.elRef0_state_0, changes["elRef0_state_0"]?.currentValue);
            this.elRef0_state_1 = {};
            this.setAttributes(this.elRef0?.nativeElement, this.elRef0_state_1, changes["elRef0_state_1"]?.currentValue);
            this.setAttributes(this.elRef0?.nativeElement, this.attributes, changes["attributes"]?.currentValue);
        }
    }
    ngOnDestroy() {
        for (const fn of this._listenerFns.values()) {
            fn();
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: FormComponent, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: FormComponent, isStandalone: true, selector: "form-component", inputs: { builderContext: "builderContext", previewState: "previewState", sendWithJs: "sendWithJs", sendSubmissionsTo: "sendSubmissionsTo", action: "action", customHeaders: "customHeaders", contentType: "contentType", sendSubmissionsToEmail: "sendSubmissionsToEmail", name: "name", method: "method", errorMessagePath: "errorMessagePath", resetFormOnSubmit: "resetFormOnSubmit", successUrl: "successUrl", validate: "validate", attributes: "attributes", errorMessage: "errorMessage", sendingMessage: "sendingMessage", successMessage: "successMessage" }, viewQueries: [{ propertyName: "formRef", first: true, predicate: ["formRef"], descendants: true }, { propertyName: "elRef0", first: true, predicate: ["elRef0"], descendants: true }], usesOnChanges: true, ngImport: i0, template: `
    <form
      [attr.validate]="validate"
      #formRef
      [attr.action]="!sendWithJs && action"
      [attr.method]="method"
      [attr.name]="name"
      (submit)="onSubmit($event)"
      #elRef0
    >
      <ng-content></ng-content>
      <ng-container *ngIf="node_1_Show">
        <blocks
          path="errorMessage"
          [blocks]="errorMessage!"
          [context]="builderContext"
        ></blocks>
      </ng-container>
      <ng-container *ngIf="node_2_Show">
        <blocks
          path="sendingMessage"
          [blocks]="sendingMessage!"
          [context]="builderContext"
        ></blocks>
      </ng-container>
      <ng-container *ngIf="node_3_Show">
        <pre class="builder-form-error-text pre">{{node_4_div}}</pre>
      </ng-container>
      <ng-container *ngIf="node_5_Show">
        <blocks
          path="successMessage"
          [blocks]="successMessage!"
          [context]="builderContext"
        ></blocks>
      </ng-container>
    </form>
  `, isInline: true, styles: [":host{display:contents}.pre{padding:10px;color:red;text-align:center}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "component", type: Blocks, selector: "blocks", inputs: ["blocks", "parent", "path", "styleProp", "className", "context", "linkComponent", "registeredComponents"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: FormComponent, decorators: [{
            type: Component,
            args: [{ selector: "form-component", template: `
    <form
      [attr.validate]="validate"
      #formRef
      [attr.action]="!sendWithJs && action"
      [attr.method]="method"
      [attr.name]="name"
      (submit)="onSubmit($event)"
      #elRef0
    >
      <ng-content></ng-content>
      <ng-container *ngIf="node_1_Show">
        <blocks
          path="errorMessage"
          [blocks]="errorMessage!"
          [context]="builderContext"
        ></blocks>
      </ng-container>
      <ng-container *ngIf="node_2_Show">
        <blocks
          path="sendingMessage"
          [blocks]="sendingMessage!"
          [context]="builderContext"
        ></blocks>
      </ng-container>
      <ng-container *ngIf="node_3_Show">
        <pre class="builder-form-error-text pre">{{node_4_div}}</pre>
      </ng-container>
      <ng-container *ngIf="node_5_Show">
        <blocks
          path="successMessage"
          [blocks]="successMessage!"
          [context]="builderContext"
        ></blocks>
      </ng-container>
    </form>
  `, standalone: true, imports: [CommonModule, Blocks], styles: [":host{display:contents}.pre{padding:10px;color:red;text-align:center}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { builderContext: [{
                type: Input
            }], previewState: [{
                type: Input
            }], sendWithJs: [{
                type: Input
            }], sendSubmissionsTo: [{
                type: Input
            }], action: [{
                type: Input
            }], customHeaders: [{
                type: Input
            }], contentType: [{
                type: Input
            }], sendSubmissionsToEmail: [{
                type: Input
            }], name: [{
                type: Input
            }], method: [{
                type: Input
            }], errorMessagePath: [{
                type: Input
            }], resetFormOnSubmit: [{
                type: Input
            }], successUrl: [{
                type: Input
            }], validate: [{
                type: Input
            }], attributes: [{
                type: Input
            }], errorMessage: [{
                type: Input
            }], sendingMessage: [{
                type: Input
            }], successMessage: [{
                type: Input
            }], formRef: [{
                type: ViewChild,
                args: ["formRef"]
            }], elRef0: [{
                type: ViewChild,
                args: ["elRef0"]
            }] } });
