// @builder.io/sdk-angular/lib/browser/esm2022/blocks/custom-code/custom-code.mjs
import { ChangeDetectionStrategy } from "@angular/core";
import { Component, ViewChild, Input, } from "@angular/core";
import { CommonModule } from "@angular/common";
import * as i0 from "@angular/core";
import * as i1 from "@angular/platform-browser";
export default class CustomCode {
    ngOnInit() {
        this.node_0_div =
            "builder-custom-code" + (this.replaceNodes ? " replace-nodes" : "");
        if (typeof window !== "undefined") {
            // TODO: Move this function to standalone one in '@builder.io/utils'
            if (!this.elementRef.nativeElement?.getElementsByTagName ||
                typeof window === "undefined") {
                return;
            }
            const scripts = this.elementRef.nativeElement.getElementsByTagName("script");
            for (let i = 0; i < scripts.length; i++) {
                const script = scripts[i];
                if (script.src) {
                    if (this.scriptsInserted.includes(script.src)) {
                        continue;
                    }
                    this.scriptsInserted.push(script.src);
                    const newScript = document.createElement("script");
                    newScript.async = true;
                    newScript.src = script.src;
                    document.head.appendChild(newScript);
                }
                else if (!script.type ||
                    [
                        "text/javascript",
                        "application/javascript",
                        "application/ecmascript",
                    ].includes(script.type)) {
                    if (this.scriptsRun.includes(script.innerText)) {
                        continue;
                    }
                    try {
                        this.scriptsRun.push(script.innerText);
                        new Function(script.innerText)();
                    }
                    catch (error) {
                        console.warn("`CustomCode`: Error running script:", error);
                    }
                }
            }
        }
    }
    constructor(sanitizer) {
        this.sanitizer = sanitizer;
        this.scriptsInserted = [];
        this.scriptsRun = [];
        this.node_0_div = null;
    }
    ngOnChanges(changes) {
        if (typeof window !== "undefined") {
            this.node_0_div =
                "builder-custom-code" + (this.replaceNodes ? " replace-nodes" : "");
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: CustomCode, deps: [{ token: i1.DomSanitizer }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: CustomCode, isStandalone: true, selector: "custom-code", inputs: { replaceNodes: "replaceNodes", code: "code" }, viewQueries: [{ propertyName: "elementRef", first: true, predicate: ["elementRef"], descendants: true }], usesOnChanges: true, ngImport: i0, template: `
    <div #elementRef [class]="node_0_div" [innerHTML]="sanitizer.bypassSecurityTrustHtml(code)"></div>
  `, isInline: true, styles: [":host{display:contents}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: CustomCode, decorators: [{
            type: Component,
            args: [{ changeDetection: ChangeDetectionStrategy.OnPush, selector: "custom-code", template: `
    <div #elementRef [class]="node_0_div" [innerHTML]="sanitizer.bypassSecurityTrustHtml(code)"></div>
  `, standalone: true, imports: [CommonModule], styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i1.DomSanitizer }]; }, propDecorators: { replaceNodes: [{
                type: Input
            }], code: [{
                type: Input
            }], elementRef: [{
                type: ViewChild,
                args: ["elementRef"]
            }] } });
