// @builder.io/sdk-angular/lib/browser/esm2022/components/block/components/block-wrapper.mjs
import { Component, Input } from "@angular/core";
import { CommonModule } from "@angular/common";
import { getBlockActions } from "../../../functions/get-block-actions";
import { getBlockProperties } from "../../../functions/get-block-properties";
import DynamicRenderer from "../../dynamic-renderer/dynamic-renderer";
import * as i0 from "@angular/core";
export default class BlockWrapper {
    constructor() {
        this.node_0_DynamicRenderer = null;
        this.node_1_DynamicRenderer = null;
    }
    ngOnInit() {
        this.node_0_DynamicRenderer = getBlockProperties({
            block: this.block,
            context: this.context,
        });
        this.node_1_DynamicRenderer = getBlockActions({
            block: this.block,
            rootState: this.context.rootState,
            rootSetState: this.context.rootSetState,
            localState: this.context.localState,
            context: this.context.context,
            stripPrefix: true,
        });
    }
    ngOnChanges(changes) {
        if (typeof window !== "undefined") {
            this.node_0_DynamicRenderer = getBlockProperties({
                block: this.block,
                context: this.context,
            });
            this.node_1_DynamicRenderer = getBlockActions({
                block: this.block,
                rootState: this.context.rootState,
                rootSetState: this.context.rootSetState,
                localState: this.context.localState,
                context: this.context.context,
                stripPrefix: true,
            });
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BlockWrapper, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: BlockWrapper, isStandalone: true, selector: "block-wrapper", inputs: { block: "block", context: "context", Wrapper: "Wrapper" }, usesOnChanges: true, ngImport: i0, template: `
    <dynamic-renderer
      [TagName]="Wrapper"
      [attributes]="node_0_DynamicRenderer"
      [actionAttributes]="node_1_DynamicRenderer"
    >
      <ng-content></ng-content>
    </dynamic-renderer>
  `, isInline: true, styles: [":host{display:contents}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "component", type: DynamicRenderer, selector: "dynamic-renderer, DynamicRenderer", inputs: ["TagName", "attributes", "actionAttributes"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BlockWrapper, decorators: [{
            type: Component,
            args: [{ selector: "block-wrapper", template: `
    <dynamic-renderer
      [TagName]="Wrapper"
      [attributes]="node_0_DynamicRenderer"
      [actionAttributes]="node_1_DynamicRenderer"
    >
      <ng-content></ng-content>
    </dynamic-renderer>
  `, standalone: true, imports: [CommonModule, DynamicRenderer], styles: [":host{display:contents}\n"] }]
        }], propDecorators: { block: [{
                type: Input
            }], context: [{
                type: Input
            }], Wrapper: [{
                type: Input
            }] } });
