// @builder.io/sdk-angular/lib/browser/esm2022/blocks/embed/embed.mjs
import { ChangeDetectionStrategy } from "@angular/core";
import { Component, ViewChild, Input, } from "@angular/core";
import { CommonModule } from "@angular/common";
import { isJsScript } from "./helpers";
import * as i0 from "@angular/core";
import * as i1 from "@angular/platform-browser";
export default class BuilderEmbed {
    findAndRunScripts() {
        if (!this.elem.nativeElement ||
            !this.elem.nativeElement.getElementsByTagName)
            return;
        const scripts = this.elem.nativeElement.getElementsByTagName("script");
        for (let i = 0; i < scripts.length; i++) {
            const script = scripts[i];
            if (script.src && !this.scriptsInserted.includes(script.src)) {
                this.scriptsInserted.push(script.src);
                const newScript = document.createElement("script");
                newScript.async = true;
                newScript.src = script.src;
                document.head.appendChild(newScript);
            }
            else if (isJsScript(script) &&
                !this.scriptsRun.includes(script.innerText)) {
                try {
                    this.scriptsRun.push(script.innerText);
                    new Function(script.innerText)();
                }
                catch (error) {
                    console.warn("`Embed`: Error running script:", error);
                }
            }
        }
    }
    constructor(sanitizer) {
        this.sanitizer = sanitizer;
        this.scriptsInserted = [];
        this.scriptsRun = [];
        this.ranInitFn = false;
    }
    ngOnChanges(changes) {
        if (typeof window !== "undefined") {
            if (this.elem.nativeElement && !this.ranInitFn) {
                this.ranInitFn = true;
                this.findAndRunScripts();
            }
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderEmbed, deps: [{ token: i1.DomSanitizer }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: BuilderEmbed, isStandalone: true, selector: "builder-embed", inputs: { content: "content" }, viewQueries: [{ propertyName: "elem", first: true, predicate: ["elem"], descendants: true }], usesOnChanges: true, ngImport: i0, template: `
    <div class="builder-embed" #elem [innerHTML]="sanitizer.bypassSecurityTrustHtml(content)"></div>
  `, isInline: true, styles: [":host{display:contents}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderEmbed, decorators: [{
            type: Component,
            args: [{ changeDetection: ChangeDetectionStrategy.OnPush, selector: "builder-embed", template: `
    <div class="builder-embed" #elem [innerHTML]="sanitizer.bypassSecurityTrustHtml(content)"></div>
  `, standalone: true, imports: [CommonModule], styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i1.DomSanitizer }]; }, propDecorators: { content: [{
                type: Input
            }], elem: [{
                type: ViewChild,
                args: ["elem"]
            }] } });
