// @builder.io/sdk-angular/lib/node/esm2022/blocks/symbol/symbol.mjs
import { Component, ViewChild, Input, } from "@angular/core";
import { CommonModule } from "@angular/common";
import ContentVariants from "../../components/content-variants/index";
import DynamicDiv from "../../components/dynamic-div";
import { getClassPropName } from "../../functions/get-class-prop-name";
import { fetchSymbolContent } from "./symbol.helpers";
import * as i0 from "@angular/core";
export default class BuilderSymbol {
    get blocksWrapper() {
        return DynamicDiv;
    }
    get contentWrapper() {
        return DynamicDiv;
    }
    get className() {
        return [
            ...[this.attributes[getClassPropName()]],
            "builder-symbol",
            this.symbol?.inline ? "builder-inline-symbol" : undefined,
            this.symbol?.dynamic || this.dynamic
                ? "builder-dynamic-symbol"
                : undefined,
        ]
            .filter(Boolean)
            .join(" ");
    }
    setContent() {
        if (this.contentToUse)
            return;
        fetchSymbolContent({
            symbol: this.symbol,
            builderContextValue: this.builderContext,
        }).then((newContent) => {
            if (newContent) {
                this.contentToUse = newContent;
            }
        });
    }
    setAttributes(el, value, changes) {
        if (!el) {
            return;
        }
        const target = typeof changes === "undefined" ? value : changes;
        Object.keys(target).forEach((key) => {
            if (key.startsWith("on")) {
                if (this._listenerFns.has(key)) {
                    this._listenerFns.get(key)();
                }
                this._listenerFns.set(key, this.renderer.listen(el, key.replace("on", "").toLowerCase(), target[key]));
            }
            else {
                this.renderer.setAttribute(el, key.toLowerCase(), target[key] ?? "");
            }
        });
    }
    constructor(renderer) {
        this.renderer = renderer;
        this._listenerFns = new Map();
        this.contentToUse = null;
        this.node_0_ContentVariants = null;
        this.node_1_ContentVariants = null;
        this.node_2_ContentVariants = null;
        this.elRef0_state_0 = null;
        this.elRef0_state_1 = null;
    }
    ngOnInit() {
        this.contentToUse = this.symbol?.content;
        this.node_0_ContentVariants = {
            ...this.builderContext.context,
            symbolId: this.builderBlock?.id,
        };
        this.node_1_ContentVariants = Object.values(this.builderComponents);
        this.node_2_ContentVariants = {
            ...this.symbol?.data,
            ...this.builderContext.localState,
            ...this.contentToUse?.data?.state,
        };
        this.elRef0_state_0 = {};
        this.elRef0_state_1 = {};
        if (typeof window !== "undefined") {
        }
    }
    ngAfterViewInit() {
        this.setAttributes(this.elRef0?.nativeElement, this.elRef0_state_0);
        this.setAttributes(this.elRef0?.nativeElement, this.attributes);
        this.setAttributes(this.elRef0?.nativeElement, this.elRef0_state_1);
    }
    ngOnChanges(changes) {
        if (typeof window !== "undefined") {
            if (changes.symbol) {
                this.setContent();
            }
            this.node_0_ContentVariants = {
                ...this.builderContext.context,
                symbolId: this.builderBlock?.id,
            };
            this.node_1_ContentVariants = Object.values(this.builderComponents);
            this.node_2_ContentVariants = {
                ...this.symbol?.data,
                ...this.builderContext.localState,
                ...this.contentToUse?.data?.state,
            };
            this.elRef0_state_0 = {};
            this.setAttributes(this.elRef0?.nativeElement, this.elRef0_state_0, changes["elRef0_state_0"]?.currentValue);
            this.setAttributes(this.elRef0?.nativeElement, this.attributes, changes["attributes"]?.currentValue);
            this.elRef0_state_1 = {};
            this.setAttributes(this.elRef0?.nativeElement, this.elRef0_state_1, changes["elRef0_state_1"]?.currentValue);
        }
    }
    ngOnDestroy() {
        for (const fn of this._listenerFns.values()) {
            fn();
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderSymbol, deps: [{ token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: BuilderSymbol, isStandalone: true, selector: "builder-symbol", inputs: { symbol: "symbol", builderContext: "builderContext", builderBlock: "builderBlock", builderComponents: "builderComponents", attributes: "attributes", dynamic: "dynamic", builderLinkComponent: "builderLinkComponent" }, viewQueries: [{ propertyName: "elRef0", first: true, predicate: ["elRef0"], descendants: true }], usesOnChanges: true, ngImport: i0, template: `
    <div [class]="className" #elRef0>
      <content-variants
        [nonce]="builderContext.nonce"
        [isNestedRender]="true"
        [apiVersion]="builderContext.apiVersion"
        [apiKey]="builderContext.apiKey!"
        [context]="node_0_ContentVariants"
        [customComponents]="node_1_ContentVariants"
        [data]="node_2_ContentVariants"
        [canTrack]="builderContext.canTrack"
        [model]="symbol?.model ?? ''"
        [content]="contentToUse"
        [linkComponent]="builderLinkComponent"
        [blocksWrapper]="blocksWrapper"
        [contentWrapper]="contentWrapper"
      ></content-variants>
    </div>
  `, isInline: true, styles: [":host{display:contents}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "component", type: ContentVariants, selector: "builder-content, content-variants", inputs: ["canTrack", "content", "isNestedRender", "nonce", "apiHost", "model", "data", "context", "apiKey", "apiVersion", "customComponents", "linkComponent", "locale", "enrich", "blocksWrapper", "blocksWrapperProps", "contentWrapper", "contentWrapperProps", "trustedHosts"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BuilderSymbol, decorators: [{
            type: Component,
            args: [{ selector: "builder-symbol", template: `
    <div [class]="className" #elRef0>
      <content-variants
        [nonce]="builderContext.nonce"
        [isNestedRender]="true"
        [apiVersion]="builderContext.apiVersion"
        [apiKey]="builderContext.apiKey!"
        [context]="node_0_ContentVariants"
        [customComponents]="node_1_ContentVariants"
        [data]="node_2_ContentVariants"
        [canTrack]="builderContext.canTrack"
        [model]="symbol?.model ?? ''"
        [content]="contentToUse"
        [linkComponent]="builderLinkComponent"
        [blocksWrapper]="blocksWrapper"
        [contentWrapper]="contentWrapper"
      ></content-variants>
    </div>
  `, standalone: true, imports: [CommonModule, ContentVariants], styles: [":host{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }]; }, propDecorators: { symbol: [{
                type: Input
            }], builderContext: [{
                type: Input
            }], builderBlock: [{
                type: Input
            }], builderComponents: [{
                type: Input
            }], attributes: [{
                type: Input
            }], dynamic: [{
                type: Input
            }], builderLinkComponent: [{
                type: Input
            }], elRef0: [{
                type: ViewChild,
                args: ["elRef0"]
            }] } });
