// @builder.io/sdk-angular/lib/node/esm2022/blocks/personalization-container/helpers.mjs
import { TARGET } from '../../constants/target';
import { isBrowser } from '../../functions/is-browser';
import { isEditing } from '../../functions/is-editing';
import { FILTER_WITH_CUSTOM_TARGETING_SCRIPT, PERSONALIZATION_SCRIPT } from './helpers/inlined-fns';
export function checkShouldRenderVariants(variants, canTrack) {
    const hasVariants = variants && variants.length > 0;
    if (TARGET === 'reactNative')
        return false;
    if (!hasVariants)
        return false;
    if (!canTrack)
        return false;
    if (TARGET === 'vue' || TARGET === 'svelte')
        return true;
    if (isBrowser())
        return false;
    return true;
}
export function getBlocksToRender({ variants, previewingIndex, isHydrated, filteredVariants, fallbackBlocks }) {
    const fallback = {
        blocks: fallbackBlocks ?? [],
        path: 'this.children'
    };
    if (isHydrated && isEditing()) {
        // If editing a specific variant
        if (typeof previewingIndex === 'number' && previewingIndex < (variants?.length ?? 0)) {
            const variant = variants[previewingIndex];
            return {
                blocks: variant.blocks,
                path: `component.options.variants.${previewingIndex}.blocks`
            };
        }
        // Otherwise we're editing the default variant
        return fallback;
    }
    // If we're on the browser, check if there's a winning variant
    if (isBrowser()) {
        const winningVariant = filteredVariants?.[0];
        if (winningVariant) {
            return {
                blocks: winningVariant.blocks,
                path: `component.options.variants.${variants?.indexOf(winningVariant)}.blocks`
            };
        }
    }
    // If no winning variant or we are on the server, return the default variant
    return fallback;
}
export const getPersonalizationScript = (variants, blockId, locale) => {
    return `
  (function() {
    ${FILTER_WITH_CUSTOM_TARGETING_SCRIPT}
    ${PERSONALIZATION_SCRIPT}
    getPersonalizedVariant(${JSON.stringify(variants)}, "${blockId}"${locale ? `, "${locale}"` : ''})
  })();
  `;
};
export { filterWithCustomTargeting } from './helpers/inlined-fns';
